src/gt4py/next/program_processors/runners/dace_fieldview/gtir_sdfg.py

```diff
@@ -671,30 +654,25 @@ def get_field_domain_offset(
 
         # populate mapping from field name to domain offset
         lambda_field_offsets: dict[str, Optional[list[dace.symbolic.SymExpr]]] = {}
-        for p_name, p_type in lambda_symbols.items():
+        for p_name, p_type in lambda_global_symbols.items():
             lambda_field_offsets |= get_field_domain_offset(p_name, p_type)
 
         # lower let-statement lambda node as a nested SDFG
-        lambda_translator = GTIRToSDFG(
-            self.offset_provider_type, lambda_symbols, lambda_field_offsets
-        )
-        nsdfg = dace.SDFG(name=self.unique_nsdfg_name(sdfg, "lambda"))
-        nstate = nsdfg.add_state("lambda")
+        nsdfg = dace.SDFG(name=self.unique_nsdfg_name("lambda"))
+        nstate = nsdfg.add_state("body")
 
-        # add sdfg storage for the symbols that need to be passed as input parameters
         lambda_params = [
-            gtir.Sym(id=p_name, type=p_type) for p_name, p_type in lambda_symbols.items()
+            gtir.Sym(id=p_name, type=p_type) for p_name, p_type in lambda_global_symbols.items()
         ]
         lambda_domain_symbols = _collect_symbols_in_domain_expressions(node.expr, lambda_params)
-        lambda_translator._add_sdfg_params(
-            nsdfg, node_params=lambda_params, symbolic_arguments=lambda_domain_symbols
-        )
-
-        lambda_result = lambda_translator.visit(
-            node.expr,
-            sdfg=nsdfg,
-            head_state=nstate,
+        lambda_translator = GTIRToSDFG(
+            nsdfg,
+            self.offset_provider_type,
```

Comment:
Why do you no longer have to pass `lambda_params` to the function.

Reply:
Because `lambda_params` corresponds to the set of keys in`lambda_global_symbols`.